proposals/0351-regex-builder.md

```diff
@@ -644,14 +644,14 @@ let regex = Regex {
   Capture(b)
 }
 
-if let result = regex.firstMatch(in: "abcdefabcdef") {
+if let result = try regex.firstMatch(in: "abcdefabcdef") {
   print(result[a]) // => Optional("abc")
   print(result[b]) // => Optional("def")
   print(result[c]) // => nil
 }
 ```
 
-A regex is considered invalid when it contains a use of reference without it ever being captured as (used as `Capture(..., as: ...)` or `Capture(as: ...) { ... }`) in the regex. When this occurs in the regex builder DSL, a runtime error will be reported. Similarly, the argument to a `Regex.Match.subscript(_:)` must have been captured as in the regex that produced the match.
+A regex is considered invalid when it contains a use of reference without it ever being used as the `as:` argument to an initializer of `Capture` or `TryCapture` in the regex. When this occurs in the regex builder DSL, a runtime error will be reported. Similarly, the argument to a `Regex.Match.subscript(_:)` must have been used as the `as:` argument to an initializer of `Capture` or `TryCapture` in the regex that produced the match.
```

Comment:
Might be worth adding a sentence or breaking into two paragraphs. It took a few glances for me to remember that the subscript always returns optional, so it's good to call that out explicitly outside of details